waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -1433,7 +1448,14 @@
         fileserver-predicate-fn (-> fileserver :predicate-fn)
         fileserver-enabled? (fileserver-predicate-fn scheduler service-id service-description context)
         fileserver-label (if fileserver-enabled? "enabled" "disabled")
-        waiter-config-token (retrieve-unique-service-mapping-token service-description)]
+        waiter-config-token (retrieve-unique-service-mapping-token service-description)
+        ;; Services that are in bypass will need to safely scale down. Configuring 'preStop' will delay the sigterm for each contianer.
+        bypass-enabled? (sd/service-description-bypass-enabled? service-description)
+        add-lifecycle-configs-fn (fn add-lifecycle-config [container-configs]
+                                   (mapv #(as-> % container-config
+                                            (assoc-in container-config [:lifecycle :preStop :exec :command] pod-bypass-pre-stop-cmd)
+                                            (update container-config :env conj {:name "WAITER_BYPASS_FORCE_SIGTERM_SECS" :value (str pod-bypass-force-sigterm-secs)}))
+                                         container-configs))]
```

Comment:
Let's move this outside into a function of its own. That allows testing this function.